doc/specification/language-proposal.md

```diff
@@ -0,0 +1,1518 @@
+| `tan(x: real)`     | Tangent of _x_ in radians                               |
+| `asin(x: real)`    | Inverse sine in range [-π/2, π/2]                       |
+| `acos(x: real)`    | Inverse cosine in range [0, π]                          |
+| `atan(x: real)`    | Inverse tangent in range (-π/2, π/2)                    |
+| `exp(x: real)`     | Natural exponential e^x of _x_                          |
+| `expm1(x: real)`   | *e*^*x* - 1 (for small *x*)                             |
+| `exprel(x: real)`  | Relative exponential ₁F₁(1; 2; *x*) = (*e*^*x* - 1)/*x* |
+| `exprelr(x: real)` | Reciprocal relative exponential 1/₁F₁(1; 2; *x*)        |
+| `log(x: real)`     | Natural logarithm of _x_                                |
+| `logp1(x: real)`   | log(*x* + 1) (for small *x*)                            |
+| `sinh(x: real)`    | Hyperbolic sine of _x_                                  |
+| `cosh(x: real)`    | Hyperbolic cosine of _x_                                |
+| `tanh(x: real)`    | Hyperbolic tangent of _x_                               |
+| `asinh(x: real)`   | Inverse hyperbolic sine  of _x_                         |
+| `acosh(x: real)`   | Inverse hyperbolic cosine  of _x_                       |
+| `atanh(x: real)`   | Inverse hyperbolic tangent of _x_                       |
+
+In addition, it provides the function `nernst(z: real, T: temperature, Cinner: concentration, Couter: concentration)` defined as _R_·_T_/_z_·(log _Cinner_ - log _Couter_), where _R_ is the molar gas constant as defined in the 2019 SI units as exactly 8.314'462'618'153'24 J/K/mol.
+
+## Modules and interfaces
+<a id="modules-and-interfaces"/>
+
+Modules are used to collect parameters, constants, and function definitions; interfaces are used to define a particular sort of Arbor functionality, such as ion channel or gap junction dynamics. Module and interface definitions can only be provided at top level.
+
+
+### Module definition
+<a id="module-definition"/>
+
+> _module-defn_ ::= `module` ***symbol*** `{` _module-decl_* `}`
+>
+> _module-decl_ ::=  _type-alias_ | _parameter-defn_ | _constant-defn_ | _function-defn_ | _module-import_
+>
+> _module-import_ ::= `import` ***symbol*** ( `as` ***symbol*** )
+>
+> _type-alias_ ::= `type` ***symbol*** `=` _type-expr_ `;`
+>
+> _parameter-defn_ ::= `parameter` ***symbol*** _type-assertion_? `=` _expression_ `;`
+>
+> _constant-defn_ ::= `def` ***symbol*** _type-assertion_? `=` _expression_ `;`
+>
+> _function-defn_ ::= `def` ***symbol*** `=` _function-literal_ `;`
+
+#### Module import
+<a id="module-imports"/>
+
+Identifiers bound in module scope in one module can be used in another module or interface, if the module is imported. If a module `A` is imported as `B`, a type, parameter, constant, or function _x_ defined in `A` can be referenced with the qualified identifier `B`._x_.
+
+Examples:
+```
+module A {
+    constant real pi = 3;
+}
+
+module X {
+    import module A;
+    constant real pi = A.pi; # Also 3.
```

Comment:
We're allowed to bind _constant expressions_ to constants (i.e. with `def` when it's not a function definition), but _parameter-constant expressions_ to parameters. This is stated a little bit below under _Parameters_ and _Constant and function definitions_.

Reply:
In
```
module X {
    import module A;
    def real pi = A.pi;
}
```
the identifier `pi` in the module scope of `X`, or the qualified identifier `X.pi` outside the module, is bound to the expression `A.pi`, which is a qualified identifier with `A` a module import for the module 'A', so that `A.pi` itself is bound to what `pi` is bound to in the module 'A'.

It's legal because everything down the chain is a _constant-expression_.